### PR TITLE
Switch apiGroup to cluster.x-k8s.io

### LINKED ISSUE #918: Switch kubebuilder domain to cluster.x-k8s.io
/kind feature

**Describe the solution you'd like**
The kubebuilder domain and generated APIs should fall under `cluster.x-k8s.io` domain.

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #918 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by vincepri** [comment]
/hold

Waiting for https://github.com/kubernetes-sigs/cluster-api/pull/1182

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -7,10 +7,10 @@ metadata:
   name: manager-role
 rules:
 - apiGroups:
-  - awsprovider.k8s.io
```
Comment (detiber) [inline config/rbac/role.yaml]: I really wish there was consistent sorting for the generation of this file :joy:

Comment (vincepri) [reply config/rbac/role.yaml]: It should have been fixed in the latest controller-tools

Comment (vincepri) [reply config/rbac/role.yaml]: https://github.com/kubernetes-sigs/controller-tools/pull/253

**Pull request merged**